docs/tutorials/notes-app/includes/navigation.md

```diff
@@ -0,0 +1,336 @@
+
+The last piece of code in the previous snippet is the `notesCollection_SelectionChanged` handler. This method takes the currently selected item, a **Note** model, and uses its information to navigate to the `NotePage`. `GoToAsync` uses a URI string for navigation. In this case, a string is constructed that uses a query string parameter to set a property on the destination page. The interpolated string representing the URI ends up looking similar to the following string:
+
+```csharp
+NotePage?ItemId=path\on\device\XYZ.notes.txt
+```
+
+The `ItemId=` parameter is set to the file name on the device where the note is stored.
+
+Visual Studio may be indicating that the `NotePage.ItemId` property doesn't exist, which it doesn't. The next step is modifying the **Note view** to load the model based on the `ItemId` parameter that you'll create.
+
+### Query string parameters
+
+The **Note view** needs to support the query string parameter, `ItemId`. Create it now:
+
+01. In the **Solution Explorer** pane, open the _Views/NotePage.xaml.cs_ file.
+01. Add the `QueryProperty` attribute to the class keyword, providing the name of the query string property, and the class property it maps to, `ItemId` and `ItemId` respectively:
+
+    :::code language="csharp" source="../snippets/navigation/csharp/Notes/Views/NotePage.xaml.cs" id="query_prop":::
+
+01. Add a new `string` property named `ItemId`. This property calls the `LoadNote` method, passing the value of the property, which in turn, should be the file name of the note:
+
+    :::code language="csharp" source="../snippets/navigation/csharp/Notes/Views/NotePage.xaml.cs" id="itemid":::
+
+01. Replace the `SaveButton_Clicked` and `DeleteButton_Clicked` handlers with the following code:
+
+    :::code language="csharp" source="../snippets/navigation/csharp/Notes/Views/NotePage.xaml.cs" id="buttons":::
+
+    The buttons are now `async`. After they're pressed, the page navigates back to the previous page by using a URI of `..`.
+
+01. Delete the `_fileName` variable from top of the code, as it's no longer used by the class.
+
+## Modify the Shell
+
+The Shell is still loading the single note page, instead, it needs to load the **AllPages view**. Open the _AppShell.xaml_ file and change the first `ShellContent` entry to point to the `AllNotesPage` instead of `NotePage`:
+
+:::code language="xaml" source="../snippets/navigation/csharp/Notes/AppShell.xaml" highlight="13":::
+
+If you run the app now, you'll notice it crashes if you press the **Add** button, complaining that it can't navigate to `NotesPage`. Every page that can be navigated to by the Shell, needs to be registered with the Shell. The `AllNotesPage` page is automatically registered with the Shell because it was added to it in the XAML with `<ShellContent>`.
+
+Register the `NotesPage` with the shell:
```

Comment:
I'll stop commenting on "the Shell" but fundamentally "the Shell" isn't a thing :(